lib/manager/tool-versions/extract.ts

```diff
@@ -0,0 +1,84 @@
+// Based on http://asdf-vm.com/manage/configuration.html#tool-versions
+import * as NpmDatasource from '../../datasource/npm';
+import { PypiDatasource } from '../../datasource/pypi';
+import { RubyVersionDatasource } from '../../datasource/ruby-version';
+import { logger } from '../../logger';
+import { isSkipComment } from '../../util/ignore';
+import { newlineRegex, regEx } from '../../util/regex';
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+
+const whiteSpaceRegex = regEx(`(\\s+)`);
+const refString = 'ref:';
+
+// Mapping between supported dependency names and their datasources
+const supportedDeps: Record<string, string> = {
+  nodejs: NpmDatasource.id,
+  ruby: RubyVersionDatasource.id,
+  python: PypiDatasource.id,
+};
+
+export function extractPackageFile(
+  content: string,
+  _: string,
+  config: ExtractConfig
+): PackageFile | null {
+  logger.trace('tool_versions.extractPackageFile()');
+
+  const deps = content
+    .split(newlineRegex)
+    .map((rawline) => {
+      const dep: PackageDependency = {};
+      const [line, comment] = rawline.split('#').map((part) => part.trim());
+      if (isSkipComment(comment)) {
```

Comment:
What about comment only lines?

Reply:
@viceice I've added a comment only line to fixture to be sure